### PR TITLE
fix(cdc): bump debezium dependence to fix postgres cdc data loss

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
- bump debezium to 2.4.2.Final
- Fix breaking changes of connector properties 
- Fix the `OffsetBackingStore` for recovery

close https://github.com/risingwavelabs/risingwave/issues/14060

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by StrikeW** [comment]
[The connector node IT](https://github.com/risingwavelabs/risingwave/actions/runs/7298551484/job/19889741383?pr=14151) fails due to initialization error. But what strange is that I cannot reproduce the problem on local environment. @wenym1  Could you help to take a look?

Some background: I split the source unit tests to another module because debezium 2.4.2 requires antlr 4.10 runtime while spark-sql requires antlr 4.8 runtime.
```
java.lang.ExceptionInInitializerError
	at com.risingwave.tracing.TracingSlf4jImpl.event(TracingSlf4jImpl.java:30)
	at com.risingwave.tracing.TracingSlf4jAdapter.debug(TracingSlf4jAdapter.java:110)
	at org.testcontainers.utility.TestcontainersConfiguration.readProperties(TestcontainersConfiguration.java:365)
	at org.testcontainers.utility.TestcontainersConfiguration.loadConfiguration(TestcontainersConfiguration.java:347)
	at org.testcontainers.utility.TestcontainersConfiguration.getInstance(TestcontainersConfiguration.java:89)
	at org.testcontainers.utility.ConfigurationFileImageNameSubstitutor.<init>(ConfigurationFileImageNameSubstitutor.java:16)
	at org.testcontainers.utility.DefaultImageNameSubstitutor.<init>(DefaultImageNameSubstitutor.java:18)
	at org.testcontainers.utility.ImageNameSubstitutor.<clinit>(ImageNameSubstitutor.java:25)
	at org.testcontainers.images.RemoteDockerImage.<init>(RemoteDockerImage.java:41)
	at org.testcontainers.containers.GenericContainer.<init>(GenericContainer.java:242)
	at org.testcontainers.containers.JdbcDatabaseContainer.<init>(JdbcDatabaseContainer.java:62)
	at org.testcontainers.containers.MySQLContainer.<init>(MySQLContainer.java:52)
	at org.testcontainers.containers.MySQLContainer.<init>(MySQLContainer.java:48)
	at com.risingwave.connector.source.MySQLSourceTest.<clinit>(MySQLSourceTest.java:44)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: io.questdb.jar.jni.LoadException: Internal error: cannot find /risingwave/jni/linux-amd64/librisingwave_java_binding.so, broken package?
	at io.questdb.jar.jni.JarJniLoader.loadLib(JarJniLoader.java:54)
	at io.questdb.jar.jni.JarJniLoader.loadLib(JarJniLoader.java:101)
	at com.risingwave.java.binding.Binding.<clinit>(Binding.java:25)
	... 34 more
```

**Comment by tabVersion** [comment]
> Fix breaking changes of connector properties

Will this influence the compatibility from our side?

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEventConsumer.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ public class DbzCdcEventConsumer
         configs.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, true);
         jsonConverter.configure(configs);
         this.converter = jsonConverter;
+        LOG.info("sourceId={}, heartbeatTopicPrefix={}", sourceId, heartbeatTopicPrefix);
```
Comment (tabVersion) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEventConsumer.java:65]: debug?

File: java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java [hunk 2]
```diff
@@ -81,12 +85,13 @@ public void configure(WorkerConfig config) {
         }
 
         String engineName = (String) conf.get(EmbeddedEngine.ENGINE_NAME.name());
-        Converter keyConverter = new JsonConverter();
+        Map<String, String> converterConfig =
+                Collections.singletonMap(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, "false");
```
Comment (tabVersion) [inline java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java:89]: what's `singletonMap` for

File: java/connector-node/risingwave-connector-service/src/main/resources/mysql.properties [hunk 3]
```diff
@@ -17,7 +15,7 @@ database.server.id=${server.id}
 database.connectionTimeZone=+00:00
 # default heartbeat interval 60 seconds
 heartbeat.interval.ms=${debezium.heartbeat.interval.ms:-60000}
-heartbeat.topics.prefix=${debezium.heartbeat.topics.prefix:-RW_CDC_HeartBeat_}
```
Comment (tabVersion) [inline java/connector-node/risingwave-connector-service/src/main/resources/mysql.properties:20]: <i class='graphite__hidden'>[Re: lines 3 to 20]<br/><br/></i>

Oh you mean the breaking changes in the properties?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14151?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by tabVersion — APPROVED** [review]
lgtm

File: java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java (hunk 2 above)
Comment (StrikeW) [reply java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java:89]: A map only contains one k-v pair.

File: java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java [hunk 4]
```diff
@@ -81,12 +85,13 @@ public void configure(WorkerConfig config) {
...
+
+        Converter keyConverter = Instantiator.getInstance(JsonConverter.class.getName());
```
Comment (xiangjinwu) [inline java/connector-node/risingwave-source-cdc/src/main/java/com/risingwave/connector/cdc/debezium/internal/ConfigurableOffsetBackingStore.java:91]: ???

**Pull request merged**